src/opm/input/eclipse/Parser/ParserKeyword.cpp

```diff
@@ -845,26 +871,65 @@ void set_dimensions( ParserItem& item,
         }
     }
 
+    void ParserKeyword::setMatchRegexSuffix(const std::string& deckNameRegexp) {
+        try {
+            const auto sfx = std::regex { deckNameRegexp };
+            this->m_matchRegexSuffix = deckNameRegexp;
+        }
+        catch (const std::exception& e) {
+            std::cerr << fmt::format("'{}' is not a valid regular expression suffix for keyword '{}':\n"
```

Comment:
This will not end up in a log, I assume that is because this is for the code generation part? In any case, I think it would be better if it failed at this point, instead of continuing. Or would that cause other issues?

Reply:
> This will not end up in a log, I assume that is because this is for the code generation part?

Correct, this message will be emitted at the code generation stage.

>  In any case, I think it would be better if it failed at this point, instead of continuing.

That's a good idea.  Sending a message to `cerr` and continuing means that we'd likely miss the erroneous suffix.  I've pushed an update that switches to `throw`-ing an `invalid_argument` in this case, and for the existing `setMatchRegex()` function.  That leads to halting the code generation process with the message below when inserting the invalid suffix `{` in the `WELL_PROBE`:

```diff
diff --git a/src/opm/input/eclipse/share/keywords/000_Eclipse100/W/WELL_PROBE b/src/opm/input/eclipse/share/keywords/000_Eclipse100/W/WELL_PROBE
index 4c4663d4e..303f727ba 100644
--- a/src/opm/input/eclipse/share/keywords/000_Eclipse100/W/WELL_PROBE
+++ b/src/opm/input/eclipse/share/keywords/000_Eclipse100/W/WELL_PROBE
@@ -227,6 +227,7 @@
     "WPIL"
   ],
   "deck_name_regex": "WU.+|(WBHWC|WGFWC|WOFWC|WWFWC)[1-9][0-9]?|WTPR.+|WTPT.+|WTPC.+|WTIR.+|WTIT.+|WTIC.+",
+  "deck_name_regex_suffix": "{",
   "size": 1,
   "items": [
     {
```

---

```
Failed to create parserkeyword from: /home/bska/work/opm/src/opm/opm-common/src/opm/input/eclipse/share/keywords/000_Eclipse100/W/WELL_PROBE
'{' is not a valid regular expression suffix ('deck_name_regex_suffix') for keyword 'WELL_PROBE'

terminate called after throwing an instance of 'std::invalid_argument'
  what():  '{' is not a valid regular expression suffix ('deck_name_regex_suffix') for keyword 'WELL_PROBE'
```